armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -331,53 +343,101 @@ def axiallyExpandAssembly(self):
             top-most Block is is then updated to reflect any expansion/contraction.
         """
         mesh = [0.0]
-        numOfBlocks = self.linked.a.countBlocksWithFlags()
         runLog.debug(
             "Printing component expansion information (growth percentage and 'target component')"
             f"for each block in assembly {self.linked.a}."
         )
+        # expand all of the components
+        for b in self.linked.a:
+            for c in iterSolidComponents(b):
+                growFrac = self.expansionData.getExpansionFactor(c)
+                # component ndens and component heights are scaled to their respective growth factor
+                c.changeNDensByFactor(1.0 / growFrac)
+                c.zbottom = b.p.zbottom
+                c.height = growFrac * b.getHeight()
+                c.ztop = c.zbottom + c.height
+
+        # align blocks on target components
         for ib, b in enumerate(self.linked.a):
-            runLog.debug(msg=f"  Block {b}")
-            blockHeight = b.getHeight()
-            # set bottom of block equal to top of block below it
-            # if ib == 0, leave block bottom = 0.0
-            if ib > 0:
-                b.p.zbottom = self.linked.linkedBlocks[b].lower.p.ztop
-            isDummyBlock = ib == (numOfBlocks - 1)
-            if not isDummyBlock:
-                for c in iterSolidComponents(b):
-                    growFrac = self.expansionData.getExpansionFactor(c)
-                    runLog.debug(msg=f"      Component {c}, growFrac = {growFrac:.4e}")
-                    c.height = growFrac * blockHeight
-                    # align linked components
-                    if ib == 0:
-                        c.zbottom = 0.0
+            if b is not self.topMostBlock:
+                targetComp = self.expansionData.getTargetComponent(b)
+                # redefine block bounds based on target component
+                b.p.zbottom = targetComp.zbottom
+                b.p.ztop = targetComp.ztop
+                b.p.height = b.p.ztop - b.p.zbottom
+                b.clearCache()
+                b.p.z = b.p.zbottom + b.getHeight() / 2.0
+                cLinkedAbove = self.linked.linkedComponents[targetComp].upper
+                if cLinkedAbove is not None:
+                    if self.expansionData.isTargetComponent(cLinkedAbove):
+                        # the linked component in the block above is the target component for the block above.
+                        # e.g., fuel to fuel. Shift the target component in the block above up (expansion) or
+                        # down (contraction) without changing its height. In this case, component mass is conserved for
+                        # both target components.
```

Comment:
```suggestion
                        # the linked component in the block above is the target component for that block. e.g., fuel to
                        # fuel. Shift the target component up (expansion) or  down (contraction) without changing its
                        # height. In this case, component mass is conserved for both target components.
```

Reply:
Just trying to streamline the language. Take it or leave it.

Reply:
took it and ran with it https://github.com/terrapower/armi/pull/2198/commits/f80cd25c031245e61605413dcd15e70857733959